### PR TITLE
Mark UrlRewriteGraphQl, User, Vault, Webapi, Widget and Wishlist interfaces as API

### PR DESCRIPTION
**Pull request by vlmed**
### Description (*)
The following interfaces marked as API:

1. app/code/Magento/UrlRewriteGraphQl/Model/Resolver/UrlRewrite/CustomUrlLocatorInterface.php
2. app/code/Magento/User/Model/Spi/NotificationExceptionInterface.php
3. app/code/Magento/User/Model/Spi/NotificatorInterface.php
4. app/code/Magento/Vault/Block/Customer/IconInterface.php
5. app/code/Magento/Webapi/Controller/Rest/RequestProcessorInterface.php
6. app/code/Magento/Widget/Block/BlockInterface.php
7. app/code/Magento/Wishlist/Controller/IndexInterface.php
8. app/code/Magento/Wishlist/Controller/WishlistProviderInterface.php
9. app/code/Magento/Wishlist/Model/AuthenticationStateInterface.php
10. app/code/Magento/Wishlist/Model/ResourceModel/Item/Product/CollectionBuilderInterface.php
11. app/code/Magento/Wishlist/Model/Wishlist/BuyRequest/BuyRequestDataProviderInterface.php

### Fixed Issues (if relevant)

1. Fixes magento/magento2#32023

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Review by coderimus — CHANGES_REQUESTED** [review]
@vlmed thank you for the adjustments. There is one point explained to me: the @since will be set automatically during the merge. So, please, remove it :)

**Review by coderimus — CHANGES_REQUESTED** [review]
.

**Comment by coderimus** [comment]
@magento run all tests

**Comment by vlmed** [comment]
@coderimus Thanks for the explanation. `@since` sections removed

**Comment by engcom-Hotel** [comment]
@magento run all tests

**Comment by engcom-Hotel** [comment]
@magento run all tests

**Comment by engcom-Hotel** [comment]
@magento run all tests

**Comment by engcom-Hotel** [comment]
@magento run all tests

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests B2B, Functional Tests EE

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests B2B, Functional Tests EE

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests EE

**Review by hws47a — COMMENTED** [review]
Hi @vlmed,

Could you explain why two files from the ticket are missing?

Regards,
Vladimir

**Comment by engcom-Hotel** [comment]
> Hi @vlmed,
> 
> Could you explain why two files from the ticket are missing?
> 
> Regards,
> Vladimir

Hi @hws47a,
I have reverted them during static tests fix, will put them back.

**Comment by engcom-Hotel** [comment]
@magento run Static Tests

**Comment by engcom-Hotel** [comment]
Hi @vlmed,
Please check the static test.

**Comment by coderimus** [comment]
@vlmed, please, when you will find failed static tests contains this phrase `references following non-public types:` add the @api notation for the referenced class/interface. It means that the @api interface/class can rely only on the @api

**Comment by engcom-Hotel** [comment]
@magento run all tests

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests B2B, Integration Tests

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests B2B

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests B2B, Integration Tests

File: lib/internal/Magento/Framework/Webapi/Rest/Request.php [hunk 1]
```diff
@@ -11,6 +11,13 @@
 use Magento\Framework\Api\SimpleDataObjectConverter;
 use Magento\Framework\Phrase;
 
+/**
+ * Class Request
+ *
+ * @api
+ *
+ * @package Magento\Framework\Webapi\Rest
```
Comment (eduard13) [inline lib/internal/Magento/Framework/Webapi/Rest/Request.php]: Hi @engcom-Hotel, I don't think there is a need for `package` annotation.

Comment (engcom-Hotel) [reply lib/internal/Magento/Framework/Webapi/Rest/Request.php]: Has been removed.

**Comment by engcom-Hotel** [comment]
@magento run all tests

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests B2B

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests B2B

**Review by eduard13 — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @eduard13, thank you for the review.
[ENGCOM-8821](https://jira.corp.magento.com/browse/ENGCOM-8821) has been created to process this Pull Request

**Comment by gabrieldagama** [comment]
Closed in favour of https://github.com/magento/magento2/pull/32660

**Pull request closed without merging**